src/guiguts/html_tools.py

```diff
@@ -1319,22 +1321,236 @@ def run(self) -> None:
+                # zipf.write(html_file_path, os.path.basename(html_file_path))
+
+                # Ensure the images folder itself is present, even if empty
+                root_arcname = os.path.basename(images_folder_path.rstrip(os.sep)) + "/"
+                zipf.writestr(root_arcname, "")  # Add empty directory entry
+
+                # Add all files inside the images folder, preserving folder structure
+                for root, _, files in os.walk(images_folder_path):
+                    for f in files:
+                        full_path = os.path.join(root, f)
+                        arcname = os.path.relpath(
+                            full_path, start=os.path.dirname(images_folder_path)
+                        )
+                        zipf.write(full_path, arcname)
+
+            buffer.seek(0)
+            return buffer
+
+        # Post zip file to ebookmaker
+        fileobj = zip_html_and_images(file_name, os.path.join(proj_dir, "images"))
+        try:
+            response = requests.post(
+                url,
+                files={"upfile1": ("{file_base}.zip", fileobj)},
+                data={
+                    "out": "json",
+                    "make_formats": ",".join(build),
+                    "myebook": ebooknum,
+                    "mytitle": get_title(),
+                },
+                timeout=timeout,
+            )
+        except requests.exceptions.Timeout as exc:
+            report_exception(f"Initial request to {url} timed out.", exc)
+            return
+        except ConnectionError as exc:
+            report_exception(f"Initial connection error to {url}.", exc)
+            return
+        # Check if HTTP request was unsuccessful
+        try:
+            response.raise_for_status()
+        except (
+            requests.exceptions.HTTPError,
+            requests.exceptions.TooManyRedirects,
+        ) as exc:
+            report_exception(f"Initial request to {url} was unsuccessful.", exc)
+            return
+
+        result = response.json()
+        output_txt_url = result["output_log"]
+        output_dir = result["output_dir"]
+
+        # Get output.txt
+        try:
+            response = requests.get(output_txt_url, timeout=timeout)
+        except requests.exceptions.Timeout as exc:
+            report_exception("Download request for output.txt timed out.", exc)
+            return
+        except ConnectionError as exc:
+            report_exception("Connection error getting output.txt.", exc)
+            return
+        # Check if HTTP request was unsuccessful
+        try:
+            response.raise_for_status()
+        except (
+            requests.exceptions.HTTPError,
+            requests.exceptions.TooManyRedirects,
+        ) as exc:
+            report_exception("Download request for output.txt was unsuccessful.", exc)
+            return
+
+        # Process output.txt into messages in dialog
```

Comment:
Probably overkill, but if this logic is the same in both local and API (and I think it is) you could put it in a function to share it (possible even a new parent class that the local and API classes inherit from).

Reply:
I fully meant to extract that, but forgot - good catch!